db/external_sst_file_test.cc

```diff
@@ -15,8 +15,9 @@
 
 namespace rocksdb {
 
-class ExternalSSTFileTest : public DBTestBase,
-                            public ::testing::WithParamInterface<bool> {
+class ExternalSSTFileTest
+    : public DBTestBase,
+      public ::testing::WithParamInterface<std::tuple<bool, bool>> {
```

Comment:
Not yet, I can add it.

Reply:
@siying I was trying to add the following test:
```
TEST_P(ExternalSSTFileBasicTest, IngestCorruptedExternalFile) {
  int file_id = 0;
  Random64 rand(time(nullptr));
  // Generate an external SST.
  ...
  {
    // Get file size
    uint64_t file_size = 0;
    ASSERT_OK(env_->GetFileSize(file_path, &file_size));
    ASSERT_GT(file_size, 8);
    std::unique_ptr<RandomRWFile> rwfile;
    ASSERT_OK(env_->NewRandomRWFile(file_path, &rwfile, EnvOptions()));
    // Manually corrupt the file
    uint64_t offset = rand.Next() % (file_size - 8);
    char scratch[8] = {0};
    Slice buf;
    ASSERT_OK(rwfile->Read(offset, sizeof(scratch), &buf, scratch));
    scratch[0] ^= 0xff; // flip one bit
    ASSERT_OK(rwfile->Write(offset, buf));
  }

  // Ingest file.
  IngestExternalFileOptions ifo;
  ifo.verify_checksums_before_ingest = std::get<1>(GetParam());
  s = db_->IngestExternalFile({file_path}, ifo);
  // Verify s
}
```
It turns out that the test will cause the program to abort due to assertion failures, which is not idea because I don't think external file ingestion failure should cause the process to die. Maybe we can look into this in a new PR. For this PR, I'm going to simply add a test which simulates only bad block checksum.

Reply:
Added new test `ExternalSSTFileBasicTest.IngestFileWithBadBlockChecksum` in the end of `db/external_sst_file_basic_test.cc`. 
Sorry I shouldn't have mixed this change with previous commits.

Reply:
Chatted with @siying , will add a simple test in which just deterministically change the first byte of the external SST file instead of choosing a random offset.